api/types.go

```diff
@@ -310,6 +310,16 @@ type CreateRequest struct {
 	Quantization string `json:"quantization,omitempty"`
 }
 
+type TemplateRequest struct {
```

Comment:
Something I considered here was to allow for returning tokens directly similar to [HF apply chat template](https://huggingface.co/docs/transformers/main/chat_templating) and not add a `/tokenize` endpoint. 

I think it makes more sense to have the tokenize endpoint but enable tokenization to save one API call (they're also both quick calls)

Reply:
What does this represent? I can't tell from the name. Also, consider reducing it to optional fields, and making required fields (if any) arguments to "TemplateHandler".